### ISSUE TITLE
Collect at ClassifierDLApproach

### ISSUE THREAD

**Issue by willjanningCAS** [JohnSnowLabs/spark-nlp#13987]
### Is there an existing issue for this?

- [X] I have searched the existing issues and did not find a match.

### Who can help?

@maziyarpanahi 

### What are you working on?

Training a ClassifierDLApproach annotator on Bert large cased English sentence embeddings of a custom text document dataset. Training is taking place in EMR on a YARN-managed cluster. 

### Current Behavior

![killed_executors_history_server](https://github.com/JohnSnowLabs/spark-nlp/assets/145142710/9b7230c4-f9d6-4913-bafa-a035577968e0)

You can see the classifier collects all the data into the driver, and all but 4 executors are killed (due to timeout, since Spark isn't using them). 

### Expected Behavior

I would expect training to continue in a distributed fashion (each executor gets its own partition of the training data), on all of the available executors. This is possible in TensorFlow. 
I would expect not to see the collect operation.

### Steps To Reproduce

train_df = spark.read.option('header', 'true').parquet(<path to parquet here>)

document_assembler = DocumentAssembler() \
    .setInputCol("fileText") \
    .setOutputCol("document")

sentenceDetector = SentenceDetectorDLModel.pretrained()\
    .setInputCols("document")\
    .setOutputCol("sentence")

bert = BertSentenceEmbeddings.pretrained("sent_bert_large_cased", "en") \
    .setInputCols("sentence")\
    .setOutputCol("bert_sentence_embeddings")

classifier_approach = ClassifierDLApproach()\
    .setInputCols("bert_sentence_embeddings")\
    .setOutputCol("class")\
    .setLabelColumn("article_or_other")\
    .setMaxEpochs(MAX_EPOCHS)\
    .setEnableOutputLogs(True)\
    .setOutputLogsPath("logs")\
    .setDropout(DROPOUT)

pipeline = Pipeline(
    stages=[
        document_assembler,
        sentenceDetector,
        bert,
        classifier_approach])

pipeline.fit(train_df)

### Spark NLP version and Apache Spark

sparknlp version 4.0.0
spark version 3.2.1
pyspark version 3.1.2

### Type of Spark Application

spark-submit

### Java Version

1.8.0_382

### Java Home Directory

/etc/alternatives/jre

### Setup and installation

conda in EMR bootstrap script

### Operating System and Version

Amazon Linux release 2 (Karoo)

### Additional Information

This is a job training ClassifierDLApproach on Bert large sentence embeddings. For ~20 kilobytes of text it takes roughly 20 minutes to train.

**Comment by maziyarpanahi** [comment]
> I would expect training to continue in a distributed fashion (each executor gets its own partition of the training data),

Short answer: No, there is no trainable annotators in Spark NLP that is distributed. 

Longer answer:

There is no distributed training Spark NLP, all the trainable annotators only train the model in the Driver. This may be possible on TensorFlow on Python, but not the `tensorflow-java` which is for Java. And it's not something optimized nor possible via broadcast in Apache Spark. 

The only solution is to increase the size of your Driver. One way for doing this easier is to use the multiple executor setup (cluster mode) to first save the results of the pipeline up until the `ClassifierDLApproach` which will be distributed, then loading that to use it to fit() `ClassifierDLApproach`in a different pipeline which will happen on a single machine. (this at least makes it easier to adjust the size of your Driver without keep redoing the embeddings stage)

**Comment by maziyarpanahi** [comment]
PS: I forgot, when you do run the first pipeline with embeddings (BERT), you will also check the Spark UI to be sure it is well distributed, all the cores are used, there is no skew rows, the DataFrame is well partitioned, etc. https://www.johnsnowlabs.com/watch-webinar-speed-optimization-benchmarks-in-spark-nlp-3-making-the-most-of-modern-hardware/

**Comment by willjanningCAS** [comment]
great, thanks for the answer and tips
